lib/Runtime/ByteCode/OpLayoutsAsmJs.h

```diff
@@ -283,165 +283,167 @@ namespace Js
 #define Uint32x4LayoutType typename SizePolicy::RegSlotType
 #define Uint16x8LayoutType typename SizePolicy::RegSlotType
 #define Uint8x16LayoutType typename SizePolicy::RegSlotType
+#define LAYOUT_FIELDS_HELPER(x, y) x ## y
+#define LAYOUT_FIELDS_DEF(x, y) LAYOUT_FIELDS_HELPER(x, y)
 #define LAYOUT_TYPE_WMS_REG2(layout, t0, t1) \
     template <typename SizePolicy> struct OpLayoutT_##layout\
     {\
-        t0##LayoutType LAYOUT_PREFIX_##t0()0;\
-        t1##LayoutType LAYOUT_PREFIX_##t1()1;\
+        t0##LayoutType LAYOUT_FIELDS_DEF(LAYOUT_PREFIX_##t0(), 0);\
```

Comment:
() is part of the macro name and it is not going to be part of the actual field name. The tricky part was to have the number to be appended to the end of the filed name to avoid naming conflicts. 
##0 does not work as ## expects strings as operands.